go/README.md

```diff
@@ -118,3 +118,46 @@ func main() {
     }
 }
 ```
+### Custom Context
+
+If you want even greater control over the lifecycle of the HTTP requests consider providing a [context](https://pkg.go.dev/context). Contexts can be set for all requests or per request.
+
+#### Custom Context for all requests
+
+Follow the example code snippet below if you want all requests to use the same context:
```

Comment:
Need to make it very clear that the `Timeout` in the api settings/config will be ignored if a custom context is set on all requests. The developer should use context.WithTimeout() to set a timeout. 

Also should note that the custom context will also apply on the automatic token fetching request.

Reply:
Done.